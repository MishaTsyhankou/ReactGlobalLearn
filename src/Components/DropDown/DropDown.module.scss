.select {
    cursor: pointer;
    padding: 15px 40px 15px 15px;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: space-between;
    box-sizing: border-box;
    margin-left: 5px;
    font-size: 16px;
    line-height: 20px;
    text-transform: uppercase;
    border: 0;
    border-radius: 3px;
    appearance: none;
    color: #ffffff;
    background-color: #232323;
}

.dropdown {
    position: relative;
    z-index: 9;
    .icon {
        padding-top: 5px;
        margin-right: 10px;
        pointer-events: none;
        position: absolute;
        top: 13px;
        inset-inline-end: 12px;
    }
}
