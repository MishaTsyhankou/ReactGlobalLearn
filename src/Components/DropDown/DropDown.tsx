import { ChangeEvent } from 'react';
import React from 'react';
import ChevronDown from '../../images/chevron.svg';

import styles from './DropDown.module.scss';

interface DropdownProps {
    options?: {
        value: string;
        text: string;
    }[];
    selected?: string;
    id?: string;
    onChange?: (e: ChangeEvent<HTMLSelectElement>) => void;
}

export const Dropdown = ({ options, selected, onChange, id }: DropdownProps) => {
    const optionItems = options.map((option) => (
        <option key={option.value} value={option.value}>
            {option.text}
        </option>
    ));

    return (
        <div className={styles.dropdown}>
            <select className={styles.select} id={id} value={selected} onChange={onChange}>
                {optionItems}
            </select>
            <img className={styles.icon} src={ChevronDown} alt="" width={22} height={10} />
        </div>
    );
};
