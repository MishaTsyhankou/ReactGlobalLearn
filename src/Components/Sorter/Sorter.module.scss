.sortingDropdown {
    display: flex;
    align-items: center;
    justify-content: flex-end;
    margin-bottom: 0;
    padding: 0;
    gap: 16px;
}

.label {
    font-weight: 400;
    font-size: 16px;
    line-height: 20px;
    text-transform: uppercase;
    color: #ffffff;
    opacity: 0.6;
}
