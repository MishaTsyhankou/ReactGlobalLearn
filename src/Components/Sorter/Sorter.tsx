import { ChangeEvent } from 'react';
import React from 'react';
import { Dropdown } from '../DropDown/DropDown';
import { sortChanged } from '../../reducers/filtersSlice';
import { useDispatch, useSelector } from 'react-redux';
import { RootState, AppDispatch } from '../../store/store';
import { useNavigate } from 'react-router-dom';

import styles from './Sorter.module.scss';

const Sorter = ({}) => {
    const sort = useSelector((state: RootState) => state.filters.sortBy);
    const sortByParam = sort;
    const dispatch = useDispatch();
    const navigate = useNavigate();

    const sortByOptions = [
        {
            value: 'release_date',
            text: 'release date',
        },
        {
            value: 'budget',
            text: 'budget',
        },
        {
            value: 'vote_average',
            text: 'rating',
        },
        {
            value: 'vote_count',
            text: 'popularity',
        },
    ];

    const onSortSelect = (e: ChangeEvent<HTMLSelectElement>) => {
        dispatch(sortChanged(e.target.value));
        navigate(`?sortBy=${e.target.value}`);
    };

    return (
        <div className={styles.sortingDropdown}>
            <label className={styles.label} htmlFor="sortingDropdown">
                Sort by:
            </label>
            <Dropdown options={sortByOptions} id="sortingDropdown" selected={sortByParam} onChange={onSortSelect} />
        </div>
    );
};

export default Sorter;
