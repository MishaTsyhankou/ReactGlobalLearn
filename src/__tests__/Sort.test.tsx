import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Sorter from '../Components/Sorter/Sorter'
import {Dropdown} from '../Components/DropDown/DropDown'





jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn,
}));

jest.mock('react-redux', () => ({
    useDispatch: jest.fn(() => jest.fn()),
    useSelector: jest.fn(() => ({ sortBy: 'vote_average' })),
  }));



describe('<Sorter />', () => {
  const TestComponent = <Sorter />;

  it('should match snapshot', () => {
    const { container } = render(<Sorter/>);

    expect(container).toMatchSnapshot();
  });

  it('default state', () => {
    const { container } = render(<Sorter/>);

    expect(container.firstChild).toHaveTextContent('Sort by');
    expect(container.firstChild).toHaveClass('sortingDropdown')
  });

  it('Click', () => {
    const { container } = render(<Sorter/>);

    expect(container.firstChild).toHaveTextContent('Sort by');
   fireEvent.click(container)
  });

  

  
  
});