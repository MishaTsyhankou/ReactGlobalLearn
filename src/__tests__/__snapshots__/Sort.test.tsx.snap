// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sorter /> should match snapshot 1`] = `
<div>
  <div
    class="sortingDropdown"
  >
    <label
      class="label"
      for="sortingDropdown"
    >
      Sort by:
    </label>
    <div
      class="dropdown"
    >
      <select
        class="select"
        id="sortingDropdown"
      >
        <option
          value="release_date"
        >
          release date
        </option>
        <option
          value="budget"
        >
          budget
        </option>
        <option
          value="vote_average"
        >
          rating
        </option>
        <option
          value="vote_count"
        >
          popularity
        </option>
      </select>
      <img
        alt=""
        class="icon"
        height="10"
        src="[object Object]"
        width="22"
      />
    </div>
  </div>
</div>
`;
